File: Python/stack.py
#Write a menu driven program to implement Stack in Python using List
def isempty(stk):
    if stk==[]:
        return True
    else:
        return False

def push(stk,item):
    stk.append(item)
    top=len(stk)-1
    
def pop(stk):
    if isempty(stk):
        return 'Underflow'
    else:
        item=stk.pop()
        if len(stk)==0:
            top=None
        else:
            top=len(stk)-1
        return item    

def peek(stk):
    if isempty(stk):
        return "Underflow"
    else:
        top=len(stk)-1
        return(stk[top])

def display(stk):
    if isempty(stk):
        print("Stack is empty")
    else:
        for i in range(len(stk)-1,-1,-1):
            print(stk[i])

stk=[]
top=None
while True:
    print("Stack Operation:","\n1.Push","\n2.Pop","\n3.Peek","\n4.Display","\n5.Exit")
    ch=int(input("Enter your choice(1-5):"))
    if ch==1:
        item=int(input("Enter item"))
        push(stk,item)
    elif ch==2:
        item=pop(stk)
        if item=='Underflow':
            print("Stack is empty")
        else:
            print("popped item is",item)
    elif ch==3:
        item=peek(stk)
        if item=="Underflow":
            print("Stack is empty")
        else:
            print("Topmost item is ",item)
    elif ch==4:
        display(stk)
    elif ch==5:
        break
    else:
        print("Invalid Choice")
#Astha Kumari,12-E,roll-15    
        
    
    

File: java/stack.java
public class Stack<T> {
    private int maxSize;
    private int top;
    private Object[] stackArray;

    public Stack(int size) {
        this.maxSize = size;
        this.stackArray = new Object[maxSize];
        this.top = -1;
    }

    public void push(T value) {
        if (isFull()) {
            System.out.println("Stack is full, cannot push " + value);
            return;
        }
        stackArray[++top] = value;
    }

    public T pop() {
        if (isEmpty()) {
            System.out.println("Stack is empty, cannot pop");
            return null;
        }
        return (T) stackArray[top--];
    }

    @SuppressWarnings("unchecked")
    public T peek() {
        if (isEmpty()) {
            System.out.println("Stack is empty, cannot peek");
            return null;
        }
        return (T) stackArray[top];
    }

    public boolean isEmpty() {
        return (top == -1);
    }

    // Check if the stack is full
    public boolean isFull() {
        return (top == maxSize - 1);
    }

    public int size() {
        return top + 1;
    }

    public static void main(String[] args) {
        Stack<Integer> stack = new Stack<>(5);
        stack.push(1);
        stack.push(2);
        stack.push(3);
        stack.push(4);
        stack.push(5);
        
        System.out.println("Top element: " + stack.peek());
        System.out.println("Popped element: " + stack.pop());
        System.out.println("Stack size: " + stack.size());
    }
}
